src/hotspot/share/cds/filemap.cpp

```diff
@@ -1565,11 +1567,36 @@ static size_t write_bitmap(const CHeapBitMap* map, char* output, size_t offset)
   return offset + size_in_bytes;
 }
 
+// The start of the archived heap has many primitive arrays (String
+// bodies) that are not marked by the oop/ptr maps. So we must have
+// lots of leading zeros.
+size_t FileMapInfo::remove_bitmap_leading_zeros(CHeapBitMap* map) {
+  size_t old_zeros = map->find_first_set_bit(0);
+  size_t old_size = map->size_in_bytes();
+
+  // Slice and resize bitmap
+  map->truncate(old_zeros, map->size());
+
+  // We want to keep track of how many zeros were removed
+  size_t new_zeros = map->find_first_set_bit(0);
+
+  assert(new_zeros == 0, "Should have removed leading zeros");
+  assert(map->size_in_bytes() <= old_size, "Map size should have decreased");
```

Comment:
The comment doesn't match the code, since you are comparing with `<=`. I think it's safe to change to `<` as we should have removed more than 64 leading bits, due to the abundance of primitive arrays.